### PR TITLE
Feature/signal real timestamp

### PR DESCRIPTION
**Pull request by jpacov**
Adding timestamps to Recorded Signals and ProtocolAnalyzer messages
Adding timestamps to messages generated in ProtocolSniffer

Before this PR, signals and messages interpreted by protocol analyzer, had no real timestamps; just some relative time measurements between each messages in signal. Now timestamps are absolute and are saved into proto.xml files correctly for future reference.

### CODE REVIEW

**Comment by jpacov** [comment]
Physical test carried with HackRf One (Native)

**Comment by jpacov** [comment]
@jopohl Let me know if you need further information.

**Comment by jpacov** [comment]
@jopohl let me know if further changes are required ;)

**Comment by PepsConti** [comment]
Dne. It was a 2 account mixed problem. Signed with both now.

**Comment by jpacov** [comment]
Dne @jopohl . It was a problem with 2 accounts mixed. Signed with both now.

**Comment by andynoack** [comment]
@jopohl can we merge now?

File: src/urh/signalprocessing/ProtocolSniffer.py [hunk 1]
```diff
@@ -211,13 +218,16 @@ def __demodulate_data(self, data):
                                 self.signal.bits_per_symbol, self.signal.center_spacing)
 
         bit_data, pauses, bit_sample_pos = self._ppseq_to_bits(ppseq, samples_per_symbol,
-                                                               self.signal.bits_per_symbol, write_bit_sample_pos=False)
+                                                               self.signal.bits_per_symbol, write_bit_sample_pos=True)
```
Comment (jopohl) [inline src/urh/signalprocessing/ProtocolSniffer.py:221]: Why is this change needed?

**Review by jopohl — COMMENTED** [review]
It would be nice to add some unit-tests or update existing ones to verify the new timestamps work as expected. I do not have an SDR here to test recording/Protocol Sniffer so if someone from the community could test this would be appreciated.

File: src/urh/signalprocessing/ProtocolSniffer.py (hunk 1 above)
Comment (jpacov) [reply src/urh/signalprocessing/ProtocolSniffer.py:215]: So that `_ppseq_to_bits` populates `bit_sample_pos` array and we are able to calculate more accurate timestamps using bit sample position.

**Comment by jpacov** [comment]
@jopohl I tested more extensively the timestamps, and made some changes and small fixes. I also added some small unit tests for the ProtocolSniffer as you requested. Let me know if that is enough.

**Comment by jopohl** [comment]
@jpacov sorry for late reply, could you please pull in latest changes from master? I just updated the CI to support more recent Python versions and like to ensure your changes are compatible.

**Comment by jpacov** [comment]
With this MR, timestamps for decoded messages are calculated correctly (useful when exporting decoded messages).
Working for signals recorded on **Record signal** window, **Sniff protocol** window and for signals derivated from other signals (e.g. after filtering a signal).

Here an example:
image
image
image

**Comment by jpacov** [comment]
> @jpacov sorry for late reply, could you please pull in latest changes from master? I just updated the CI to support more recent Python versions and like to ensure your changes are compatible.

Done. Let me know if further actions are needed.

**Comment by jopohl** [comment]
Thank you for your contribution.
